### ISSUE TITLE
react-native pod install failed

### ISSUE THREAD

**Issue by tlimz** [facebook/react-native#38610]
### Description

I create react-native project  failed, when Installing CocoaPods dependencies.
Then I excute `cd /projectDir/ios&&pod install`  has the same error

### React Native Version

0.72.3

### Output of `npx react-native info`

info Fetching system and libraries information...
System:
  OS: macOS 12.6.2
  CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
  Memory: 54.26 MB / 8.00 GB
  Shell:
    version: 3.2.57
    path: /bin/bash
Binaries:
  Node:
    version: 16.14.2
    path: /usr/local/bin/node
  Yarn:
    version: 1.17.3
    path: /usr/local/bin/yarn
  npm:
    version: 8.5.0
    path: /usr/local/bin/npm
  Watchman:
    version: 2023.07.17.00
    path: /usr/local/bin/watchman
Managers:
  CocoaPods:
    version: 1.12.1
    path: /usr/local/opt/ruby/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.2
      - iOS 16.2
      - macOS 13.1
      - tvOS 16.1
      - watchOS 9.1
  Android SDK:
    API Levels:
      - "23"
      - "25"
      - "26"
      - "29"
      - "33"
    Build Tools:
      - 33.0.0
    System Images:
      - android-23 | Intel x86 Atom
      - android-23 | Intel x86 Atom_64
      - android-23 | Google APIs Intel x86 Atom_64
      - android-25 | Google APIs Intel x86 Atom
      - android-33 | Android TV Intel x86 Atom
      - android-33 | Google APIs Intel x86_64 Atom
    Android NDK: Not Found
IDEs:
  Android Studio: 2022.2 AI-222.4459.24.2221.10121639
  Xcode:
    version: 14.2/14C18
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.9.1
    path: /usr/bin/javac
  Ruby:
    version: 3.2.2
    path: /usr/local/opt/ruby/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found

### Steps to reproduce

`cd /projectDir/ios&&pod install`
info:

```
Framework build type is static library
[Codegen] Generating ./build/generated/ios/React-Codegen.podspec.json
Analyzing dependencies
Fetching podspec for `DoubleConversion` from `../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec`
[Codegen] Found FBReactNativeSpec
Fetching podspec for `RCT-Folly` from `../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec`
[Codegen] Found rncore
Fetching podspec for `boost` from `../node_modules/react-native/third-party-podspecs/boost.podspec`
Fetching podspec for `glog` from `../node_modules/react-native/third-party-podspecs/glog.podspec`
Fetching podspec for `hermes-engine` from `../node_modules/react-native/sdks/hermes-engine/hermes-engine.podspec`
[Hermes] Using the release tarball from Maven Central
Downloading dependencies
Installing CocoaAsyncSocket (7.6.5)
Installing DoubleConversion (1.1.6)
Installing FBLazyVector (0.72.3)
Installing FBReactNativeSpec (0.72.3)
Installing Flipper (0.182.0)
Installing Flipper-Boost-iOSX (1.76.0.1.11)
Installing Flipper-DoubleConversion (3.2.0.1)
Installing Flipper-Fmt (7.1.7)
Installing Flipper-Folly (2.6.10)
Installing Flipper-Glog (0.5.0.5)
Installing Flipper-PeerTalk (0.0.4)
Installing FlipperKit (0.182.0)
Installing OpenSSL-Universal (1.1.1100)
Installing RCT-Folly (2021.07.22.00)
Installing RCTRequired (0.72.3)
Installing RCTTypeSafety (0.72.3)
Installing React (0.72.3)
Installing React-Codegen (0.72.3)
Installing React-Core (0.72.3)
Installing React-CoreModules (0.72.3)
Installing React-NativeModulesApple (0.72.3)
Installing React-RCTActionSheet (0.72.3)
Installing React-RCTAnimation (0.72.3)
Installing React-RCTAppDelegate (0.72.3)
Installing React-RCTBlob (0.72.3)
Installing React-RCTImage (0.72.3)
Installing React-RCTLinking (0.72.3)
Installing React-RCTNetwork (0.72.3)
Installing React-RCTSettings (0.72.3)
Installing React-RCTText (0.72.3)
Installing React-RCTVibration (0.72.3)
Installing React-callinvoker (0.72.3)
Installing React-cxxreact (0.72.3)
Installing React-debug (0.72.3)
Installing React-hermes (0.72.3)
Installing React-jsi (0.72.3)
Installing React-jsiexecutor (0.72.3)
Installing React-jsinspector (0.72.3)
Installing React-logger (0.72.3)
Installing React-perflogger (0.72.3)
Installing React-rncore (0.72.3)
Installing React-runtimeexecutor (0.72.3)
Installing React-runtimescheduler (0.72.3)
Installing React-utils (0.72.3)
Installing ReactCommon (0.72.3)
Installing SocketRocket (0.6.1)
Installing Yoga (1.14.0)
Installing YogaKit (1.18.1)
Installing boost (1.76.0)
Installing fmt (6.2.1)
Installing glog (0.3.5)
[!] /bin/bash -c
set -e
#!/bin/bash
# Copyright (c) Meta Platforms, Inc. and affiliates.
#
# This source code is licensed under the MIT license found in the
# LICENSE file in the root directory of this source tree.

set -e

PLATFORM_NAME="${PLATFORM_NAME:-iphoneos}"
CURRENT_ARCH="${CURRENT_ARCH}"

if [ -z "$CURRENT_ARCH" ] || [ "$CURRENT_ARCH" == "undefined_arch" ]; then
    # Xcode 10 beta sets CURRENT_ARCH to "undefined_arch", this leads to incorrect linker arg.
    # it's better to rely on platform name as fallback because architecture differs between simulator and device

    if [[ "$PLATFORM_NAME" == *"simulator"* ]]; then
        CURRENT_ARCH="x86_64"
    else
        CURRENT_ARCH="arm64"
    fi
fi

# @lint-ignore-every TXT2 Tab Literal
if [ "$CURRENT_ARCH" == "arm64" ]; then
    cat <<\EOF >>fix_glog_0.3.5_apple_silicon.patch
diff --git a/config.sub b/config.sub
index 1761d8b..43fa2e8 100755
--- a/config.sub
+++ b/config.sub
@@ -1096,6 +1096,9 @@ case $basic_machine in
 		basic_machine=z8k-unknown
 		os=-sim
 		;;
+	arm64-*)
+		basic_machine=$(echo $basic_machine | sed 's/arm64/aarch64/')
+		;;
 	none)
 		basic_machine=none-none
 		os=-none
EOF

    patch -p1 config.sub fix_glog_0.3.5_apple_silicon.patch
fi

export CC="$(xcrun -find -sdk $PLATFORM_NAME cc) -arch $CURRENT_ARCH -isysroot $(xcrun -sdk $PLATFORM_NAME --show-sdk-path)"
export CXX="$CC"

# Remove automake symlink if it exists
if [ -h "test-driver" ]; then
    rm test-driver
fi

# Manually disable gflags include to fix issue https://github.com/facebook/react-native/issues/28446
sed -i.bak -e 's/\@ac_cv_have_libgflags\@/0/' src/glog/logging.h.in && rm src/glog/logging.h.in.bak
sed -i.bak -e 's/HAVE_LIB_GFLAGS/HAVE_LIB_GFLAGS_DISABLED/' src/config.h.in && rm src/config.h.in.bak

./configure --host arm-apple-darwin

cat << EOF >> src/config.h
/* Add in so we have Apple Target Conditionals */
#ifdef __APPLE__
#include <TargetConditionals.h>
#include <Availability.h>
#endif

/* Special configuration for ucontext */
#undef HAVE_UCONTEXT_H
#undef PC_FROM_UCONTEXT
#if defined(__x86_64__)
#define PC_FROM_UCONTEXT uc_mcontext->__ss.__rip
#elif defined(__i386__)
#define PC_FROM_UCONTEXT uc_mcontext->__ss.__eip
#endif
EOF

# Prepare exported header include
EXPORTED_INCLUDE_DIR="exported/glog"
mkdir -p exported/glog
cp -f src/glog/log_severity.h "$EXPORTED_INCLUDE_DIR/"
cp -f src/glog/logging.h "$EXPORTED_INCLUDE_DIR/"
cp -f src/glog/raw_logging.h "$EXPORTED_INCLUDE_DIR/"
cp -f src/glog/stl_logging.h "$EXPORTED_INCLUDE_DIR/"
cp -f src/glog/vlog_is_on.h "$EXPORTED_INCLUDE_DIR/"

patching file config.sub
Hunk #1 FAILED at 1096.
1 out of 1 hunk FAILED -- saving rejects to file config.sub.rej
```

I tried 
```
xcrun -k --sdk iphoneos --show-sdk-pat &&  sudo xcode-select --switch /Applications/Xcode.app
```
but it is still not works.
how can I resolve this problem?

### Snack, screenshot, or link to a repository
https://github.com/tlimz/react-native-demo

<img width="1072" alt="截屏2023-07-25 14 08 54" src="https://github.com/facebook/react-native/assets/137807220/b887232f-55b9-4ecf-a99d-f6a90a49d7ea">
<img width="1051" alt="截屏2023-07-25 14 08 34" src="https://github.com/facebook/react-native/assets/137807220/27da70e0-2d8a-4ec6-b46a-0100fccd02ce">

**Comment by tlimz** [comment]
I need help thanks

**Comment by iBotPeaches** [comment]
@tlimz - I downloaded and pulled your sample and had no issues. I'm wondering why you aren't trying to follow the official guide when it comes to dependencies/tools around ruby? Based on your command you aren't invoking Cocoapods via Bundler - so who knows what combination of things could cause the mistake you see.

See this doc - https://reactnative.dev/docs/environment-setup#optional-configuring-your-environment

Importantly the:
 * `bundle install`
 * `cd ios; docker exec pod install`
 
As when I did those steps with your example repo - I had no problem.

![Screenshot 2023-07-25 at 12 13 52 PM](https://github.com/facebook/react-native/assets/611784/7694310f-6e32-4708-8780-be3ec8acf773)
